src/wallet/EnsoWalletV2Factory.sol

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.20;
+
+import { Token, TokenType } from "../interfaces/IEnsoRouter.sol";
+
+import { IEnsoWalletV2 } from "./interfaces/IEnsoWalletV2.sol";
+
+import { IERC1155 } from "openzeppelin-contracts/token/ERC1155/IERC1155.sol";
+import { IERC20, SafeERC20 } from "openzeppelin-contracts/token/ERC20/utils/SafeERC20.sol";
+import { IERC721 } from "openzeppelin-contracts/token/ERC721/IERC721.sol";
+import { LibClone } from "solady/utils/LibClone.sol";
+
+contract EnsoWalletV2Factory {
```

Comment:
I'd create the `IEnsoWalletV2Factory` interface where to locate the errors, events and methods